### PR TITLE
Cheick/add cors support

### PR DESCRIPTION
**Pull request by cheickblp**
### Problem
Back-end server has no support for CORS requests

### Solution
Create and Register a Jersey filter class that handles `OPTIONS` pre-flight requests that are sent by the browser prior to sending cross-origin requests

### CODE REVIEW

File: api/src/main/java/marquez/api/filter/CorsFilter.java [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+     *
+     * @return allow credentials
+     */
+    public boolean isAllowCredentials() {
+        return allowCredentials;
+    }
+
+
+    /**
+     * Will allow all by default.
```
Comment (mikkel-theoremlp) [inline api/src/main/java/marquez/api/filter/CorsFilter.java:41]: If I am reading this correctly it only allows `GET` by default, is that incorrect?

File: docker-compose.yml [hunk 2]
```diff
@@ -34,7 +34,7 @@ services:
     image: postgres:12.1
     container_name: marquez-db
     ports:
-      - "5432"
+      - "5432:5432"
```
Comment (mikkel-theoremlp) [inline docker-compose.yml:37]: Why was this needed?

File: api/src/main/java/marquez/api/filter/CorsFilter.java [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+            return;
+        }
+        responseContext.getHeaders().putSingle(CorsHeaders.ACCESS_CONTROL_ALLOW_ORIGIN, origin);
+        responseContext.getHeaders().putSingle(CorsHeaders.VARY, CorsHeaders.ORIGIN);
+        if (isAllowCredentials())
+            responseContext.getHeaders().putSingle(CorsHeaders.ACCESS_CONTROL_ALLOW_CREDENTIALS, "true");
+    }
+
+
+    protected void preflight(String origin, ContainerRequestContext requestContext) throws IOException {
```
Comment (mikkel-theoremlp) [inline api/src/main/java/marquez/api/filter/CorsFilter.java:80]: Curious, this looks sane to me but how did you come up with it? 
It seems like the sort of code that would live in a very standard library.

File: api/src/main/java/marquez/MarquezApp.java [hunk 4]
```diff
@@ -176,5 +178,6 @@ private void registerFilters(@NonNull Environment env, MarquezContext marquezCon
     env.jersey()
         .getResourceConfig()
         .register(new JobRedirectFilter(marquezContext.getJobService()));
+    env.jersey().getResourceConfig().register(new CorsFilter());
```
Comment (mikkel-theoremlp) [inline api/src/main/java/marquez/MarquezApp.java:181]: Can you not register it as part of the previous line?

**Review by mikkel-theoremlp — COMMENTED** [review]
A few questions and thoughts

File: api/src/main/java/marquez/api/filter/CorsFilter.java (hunk 1 above)
Comment (cheickblp) [reply api/src/main/java/marquez/api/filter/CorsFilter.java:41]: Yes, that comment is inaccurate. The actual logic allows `GET` by default. I will fix that in the comment.

File: docker-compose.yml (hunk 2 above)
Comment (cheickblp) [reply docker-compose.yml:37]: This exposes the Postgres container on port `5432` of my machine. I needed it for connecting my development Postgres client in Pycharm to it and not have to lookup the auto-assigned port every time I re-ran the db image. This keeps the port-mapping constant.

File: api/src/main/java/marquez/api/filter/CorsFilter.java (hunk 3 above)
Comment (cheickblp) [reply api/src/main/java/marquez/api/filter/CorsFilter.java:80]: I looked up how [RestEasy](https://github.com/resteasy/Resteasy/blob/main/resteasy-core/src/main/java/org/jboss/resteasy/plugins/interceptors/CorsFilter.java#L159-L191) does it and changed it a bit

File: api/src/main/java/marquez/MarquezApp.java (hunk 4 above)
Comment (cheickblp) [reply api/src/main/java/marquez/MarquezApp.java:181]: Absolutely. I started a new line because that's what they did with the two previous registrations above so I was trying to stay consistent with that style. What I can do is call them all on the same `getResourceConfig()` so it doesn't look odd to have 2 combined and 1 by itself. It would be something like this:
```java
    env.jersey().getResourceConfig()
            .register(new LoggingMdcFilter())
            .register(new JobRedirectFilter(marquezContext.getJobService()))
            .register(new CorsFilter());
```
What do you think?

File: docker-compose.yml (hunk 2 above)
Comment (mikkel-theoremlp) [reply docker-compose.yml:37]: Understood, thank you for the context

File: api/src/main/java/marquez/api/filter/CorsFilter.java (hunk 1 above)
Comment (mikkel-theoremlp) [reply api/src/main/java/marquez/api/filter/CorsFilter.java:41]: I still see the old comment here, did you update this?

Comment (cheickblp) [reply api/src/main/java/marquez/api/filter/CorsFilter.java]: Sorry, I missed that. I just did now

**Review by mikkel-theoremlp — APPROVED** [review]
Lgtm
